drivers/sensor/lis2du12/Kconfig

```diff
@@ -0,0 +1,61 @@
+# ST Microelectronics LIS2DU12 3-axis accelerometer sensor driver
+
+# Copyright (c) 2023 STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig LIS2DU12
+	bool "LIS2DU12 I2C/SPI smartxl Chip"
+	default y
+	depends on DT_HAS_ST_LIS2DU12_ENABLED
+	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_LIS2DU12),i2c)
+	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ST_LIS2DU12),spi)
+	select HAS_STMEMSC
+	select USE_STDC_LIS2DU12
+	help
+	  Enable driver for LIS2DU12 smartxl
+	  sensor.
+
+if LIS2DU12
+
+choice LIS2DU12_TRIGGER_MODE
+	prompt "Trigger mode"
+	help
+	  Specify the type of triggering to be used by the driver.
+
+config LIS2DU12_TRIGGER_NONE
+	bool "No trigger"
+
+config LIS2DU12_TRIGGER_GLOBAL_THREAD
+	bool "Use global thread"
+	depends on GPIO
+	select LIS2DU12_TRIGGER
+
+config LIS2DU12_TRIGGER_OWN_THREAD
+	bool "Use own thread"
+	depends on GPIO
+	select LIS2DU12_TRIGGER
+
+endchoice
```

Comment:
unrelated to this PR but let me pitch the idea to you: this pattern of having an option for how to handle asyncronous events (thread, workqueue) is copypasted everywhere. How would you feel about having a Kconfig template to define these options (like the log one, the one you use with `source "subsys/logging/Kconfig.template.log_config"`) and then common set of of macro to prepare and use the support data for the selected operating mode? (thread, workqueue, syncronous or none)

Haven't quite tried to do it but I think it could remove lot of duplicated code.

Reply:
Yes, looks a great idea. Let me explore more on that.

Reply:
Thanks!

Reply:
ok, take a look just to see I did what you wanted to (not 100% sure)

Reply:
brilliant, yeah I was actually thinking about also having a set of macros to dedup the code as well but this is a brilliant start and a step towards it, thanks for doing it, let's think about the code in followups

@MaureenHelm @teburd what do you think?

Reply:
It would be great, thought to be frank I'd rather see us move in the newer sensor read API approach which avoids the need for work queues/threads at all